prefix_db.go

```diff
@@ -267,7 +286,8 @@ type prefixIterator struct {
 }
 
 func newPrefixIterator(prefix, start, end []byte, source Iterator) *prefixIterator {
-	if !source.Valid() || !bytes.HasPrefix(source.Key(), prefix) {
+	key, _ := source.Key() //nolint:errcheck
```

Comment:
reason for `nolint:errcheck` here is that it does come with an error quite often, even through the tests. 

Either I can ignore it like above, I can have this function return the error, or just print it out? Leave your opinion below while reviewing

Reply:
Why does it return an error quite often? We should figure out why and maybe alleviate such cases, or at least reduce the probability. Are these errors benign? I.e. if an error happens, will iteration still be valid? If so, I would simply ignore or add a debug statement. I'm OK with either. If it's not benign, then we need to fix the root cause.

Reply:
according to the fucntion it was treated previously as ignored and checked in `!bytes.HasPrefix(source.Key(), prefix)` so the error is benign.

this is what I am interpreting as the check if the prefix is present. It may be easier to see if i clean it up. 
```
	if !source.Valid() || !bytes.HasPrefix(key, prefix) {
		return &prefixIterator{
			prefix: prefix,
			start:  start,
			end:    end,
			source: source,
			valid:  false,
		}
	}
```